### DISCUSSION TITLE
Plotter show_bounds axes properties

### DISCUSSION THREAD

**Discussion by suargi** [pyvista/pyvista#4459] [Q&A] [answered]
Hi all,

I have the following code to create a 2d plot of a cylinder:

```python
import pyvista as pv

vtufile = 'cylinder.vtu'
reader = pv.get_reader(vtufile, force_ext='.vtu')
mesh = reader.read()

plotter = pv.Plotter(window_size=[1800, 350],
                     line_smoothing=True,
                     polygon_smoothing=True,
                     theme=pv.themes.ParaViewTheme())

plotter.add_mesh(mesh,
                 scalars='u',
                 show_scalar_bar=False,
                 #show_edges=True
                 n_colors=12,
                 )

# crop visualization field
mesh.clip(normal='x',
          origin=[-1,0,0],
          invert=False,
          inplace=True)
mesh.clip(normal='x',
          origin=[5,0,0],
          invert=True,
          inplace=True)
mesh.clip(normal='y',
          origin=[0,-1,0],
          invert=False,
          inplace=True)
mesh.clip(normal='y',
          origin=[0,1,0],
          invert=True,
          inplace=True)

# show outline edges
plotter.show_bounds(mesh=mesh,
                    axes_ranges=[-1, 5, -2, 2, 0, 0],
                    color='k',
                    show_xaxis=True,
                    show_yaxis=True,
                    show_xlabels=True,
                    show_ylabels=True,
                    bold=True,
                    xtitle='X',
                    ytitle="Y",
                    font_size=42,
                    #n_xlabels=10,
                    #n_ylabels=2,
                    location='outer',
                    use_2d=True,
                    use_3d_text=False,
                    minor_ticks=True,
                    all_edges=True,
                    ticks='inside',
                    render=True)

# Add a scalar bar
plotter.add_scalar_bar(vertical=True)

plotter.view_xy()
plotter.camera.tight(padding=0.5)
plotter.show()
```
You can get the file cylinder.vtu from [here](https://github.com/suargi/pyvista/blob/main/cylinder.vtu). The above code will produce the following result:

![image](https://github.com/pyvista/pyvista/assets/64904120/d0b1d3bb-cb5a-47e1-81b4-e78bddac91d8)

I have several questions/issues:

- The x-axis title font is not the one I have specified in plotter.show_bounds, it is smaller.
- The x-axis label font is not constant. As you can see, at the extremes the font size is correct, but it is smaller in the middle. Is it possible to correct this?
- I want to have axes ticks on bottom, top, left and right sides of the figure. However, labels and title should only appear on left and bottom. Is this possible? I have tried with different values of the parameter 'location' without success.

Thank you.

Some useful information:
- pyvista version: 0.39.1
- OS: Fedora 37

**Comment by edabor** [comment, marked as answer]
Hi @suargi !

I managed to obtain a uniform font size for the `x axis`.
![test](https://github.com/pyvista/pyvista/assets/117822458/964b9d9c-ea73-4d2f-80d6-f0018870749b)

Here are the differences with your code snippet:
![image](https://github.com/pyvista/pyvista/assets/117822458/b280acbf-f765-45c3-a923-791143b045d6)

The most important part was to call `plotter.view_xy()` after `plotter.camera.tight(padding=0.5)`.
I am not an expert in plotter and bounds so I don't know what exactly happened, but I suspect a camera issue since resetting the camera interactively made the font size uniform.
I hope it solved one of your problems.

> I want to have axes ticks on bottom, top, left and right sides of the figure. However, labels and title should only appear on left and bottom. Is this possible? I have tried with different values of the parameter 'location' without success.

As already said, I am not an expert in bounds representation so I am afraid I cannot help you for that one.
But it appears that `paraview` can show you all the axes, so you might be able to find a solution with `pyvista` by digging.

If your goal is only to look at a single result, I think your best option is to use `paraview` for its high flexibility.
![image](https://github.com/pyvista/pyvista/assets/117822458/49f6ede5-1d2a-440a-92b0-4020df3dafcc)
![image](https://github.com/pyvista/pyvista/assets/117822458/8da8e57c-c1c3-4663-bb73-3c15ceb7d94a)
Edit button of `Axes Grid` with advanced mode toggled:
![image](https://github.com/pyvista/pyvista/assets/117822458/606622c1-63e7-40ef-a666-a4ebafa98040)

**Reply by edabor** [reply]
It seems that setting `location='all'` works with my suggestion.
But it seems harder to customize compared to `paraview`.
Moreover, the padding with the data is too high (according to my taste :wink:).
![screenshot](https://github.com/pyvista/pyvista/assets/117822458/247a296e-57d3-4484-b47f-fa98f2797847)

**Reply by suargi** [reply]
Hi @beroda!

Thank you very much for the hint regarding the uniform font size. That completely solved the problem :)

Regarding the second issue, thank you once again for your input, it definitely helps. Knowing that it is indeed possible with paraview I will now dig into pyvista/paraview and try to come with a solution. Then will post it here.

**Reply by suargi** [reply]
Unfortunately it does not seem possible in paraview to have axes ticks on bottom, top, left and right sides of the figure but labels and title appearing only the on left and bottom.
Besides, I could not find a way to reduce the padding.

Thanks your the help!
